### PR TITLE
feat(2437): Support read-only SCM [6]

### PR DESCRIPTION
**Pull request by tkyi**
## Context

Users should be able to make changes to read-only pipelines. Instead of checking permission against a child pipeline, can check against parent.

## Objective

This PR:
- updates SCM config with read-only config
- passes in `readOnly` info to `auth/contexts`
- changes permissions calls to get parent permissions for read-only scm pipelines

## References

Related to https://github.com/screwdriver-cd/screwdriver/issues/2437

## License
I confirm that this contribution is made under a BSD license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: plugins/events/create.js [hunk 1]
```diff
@@ -19,270 +20,216 @@ module.exports = () => ({
...
+                return boom.notFound('No jobs to start.');
+            }
+            // everything succeeded, inform the user
+            const location = urlLib.format({
+                host: request.headers.host,
+                port: request.headers.port,
+                protocol: request.server.info.protocol,
+                pathname: `${request.path}/${event.id}`
+            });
+
```
Comment (pritamstyz4ever) [inline plugins/events/create.js:228]: Refactoring into smaller functions would be nice and make it more readable. (can update later)

**Review by pritamstyz4ever — APPROVED** [review]
check if you need to mock ''helper.js' in tests

**Comment by tkyi** [comment]
Shouldn't need to mock `helper.js` since all the underlying calls are mocked

File: plugins/events/create.js (hunk 1 above)
Comment (tkyi) [reply plugins/events/create.js:228]: I'll save that for a future PR

**Pull request merged**